docs/experiments-dpr.md

```diff
@@ -6,131 +6,436 @@ Vladimir Karpukhin, Barlas Oğuz, Sewon Min, Patrick Lewis, Ledell Wu, Sergey Ed
+                                                           --index wikipedia-dpr \
+                                                           --input runs/run.dpr.squad.multi.bf.trec \
+                                                           --output runs/run.dpr.squad.multi.bf.json
+```
+
+Evaluate
+```bash
+$ python scripts/dpr/evaluate.py --retrieval runs/run.dpr.squad.multi.bf.json --topk 20
+$ python scripts/dpr/evaluate.py --retrieval runs/run.dpr.squad.multi.bf.json --topk 100
+Top20  accuracy: 0.5198675496688742
+Top100 accuracy: 0.6772942289498581
+```
+
+### BM25 retrieval
+
+```bash
+$ python -m pyserini.search --topics squad_test_dpr \
+                             --index wikipedia-dpr \
+                             --output runs/run.squad-test.bm25.trec
+```
+
+
+To evaluate convert the TREC style run file to retrieval result file in `json` format
+```bash
+$ python -m scripts.dpr.convert_trec_run_to_retrieval_json --topics squad_test_dpr \
+                                                           --index wikipedia-dpr \
+                                                           --input runs/run.squad-test.bm25.trec \
+                                                           --output runs/run.squad-test.bm25.json
+```
+
+Evaluate
+```bash
+$ python scripts/dpr/evaluate.py --retrieval runs/run.squad-test.bm25.json --topk 20
+$ python scripts/dpr/evaluate.py --retrieval runs/run.squad-test.bm25.json --topk 100
+Top20  accuracy: 0.710879848628193
+Top100 accuracy: 0.8183538315988647
+```
+
+### Hybrid Dense-Sparse Retrieval
+Hybrid
+- dense retrieval with DPR, brute force index.
+- sparse retrieval with BM25.
+
+```bash
+$ python -m pyserini.hsearch   dense --index wikipedia-dpr-multi-bf \
+                                     --batch-size 72 --threads 72 \
+                             sparse --index wikipedia-dpr \
+                             fusion --alpha 0.24 \
+                             run  --topics squad_test_dpr \
+                                  --output runs/run.squad-test.dpr.bf.bm25.trec 
+```
+
+To evaluate convert the TREC style run file to retrieval result file in `json` format
+```bash
+$ python -m scripts.dpr.convert_trec_run_to_retrieval_json --topics squad_test_dpr \
+                                                           --index wikipedia-dpr \
+                                                           --input runs/run.squad-test.dpr.bf.bm25.trec  \
+                                                           --output runs/run.squad-test.dpr.bf.bm25.json
+```
+Evaluate
+```bash
+$ python scripts/dpr/evaluate.py --retrieval runs/run.squad-test.dpr.bf.bm25.json --topk 20
+$ python scripts/dpr/evaluate.py --retrieval runs/run.squad-test.dpr.bf.bm25.json --topk 100
+Top20  accuracy: 0.5818353831598865
+Top100 accuracy: 0.7264900662251655
+```
+
+## Summary
+| Dataset     | Method | (in paper)Top20 | (in paper)Top100 | (our) Top20        | (our) Top100       |
```

Comment:
group the cols: top 20 (paper) | top 20 (us)
same number of digits also - makes easier to compare